### ISSUE TITLE
CollectiveAllReduceStrategy RecvBufResponse error in ResNet model

### ISSUE THREAD

**Issue by CheukNgai** [tensorflow/tensorflow#23283]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): custom code
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS 10.13.6
- TensorFlow installed from (source or binary): binary(modify the code to fix the bug, pull request [link](https://github.com/tensorflow/estimator/pull/3) )
- TensorFlow version (use command below): 1.12.0-rc0
- Python version: 2.7.10
- Bazel version (if compiling from source): 0.16.1
- GCC/Compiler version (if compiling from source):  LLVM version 9.1.0(clang-902.0.39.2)
- CUDA/cuDNN version: no installed
- GPU model and memory: Intel Iris Plus Graphics

**Describe the current behavior**
using the collective_all_reduce_strategy to train cifar10_Resnet model will raise an error, but the ParameterServerStrategy won't.

**Describe the expected behavior**
This cifar10_ResNet has been updated to r1.11 and used tf.estimator.train_and_evaluate to train model, with the new feature of using distribute strategy.
It was found that the ParameterServerStrategy ran normally, but the CollectiveAllReduceStrategy raised an error in RecvBufResponse.

**Code to reproduce the issue**
Source code is attached.
[cifar10_estimator.zip](https://github.com/tensorflow/tensorflow/files/2517601/cifar10_estimator.zip)

To run the ParameterServerStrategy, type these two commands in different shells:

```
python PS_ps0.py --data-dir=./cifar-10-data --job-dir=./model_dir/ --num-gpus=0 --train-steps=1000
```

```
python PS_worker0.py --data-dir=./cifar-10-data --job-dir=./model_dir/ --num-gpus=0 --train-steps=1000
```

To run the CollectiveAllReduceStrategy(just modify TF_CONFIG and strategy), type these two commands in different shells:

```
python Collective_worker0.py --data-dir=./cifar-10-data --job-dir=./model_dir/ --num-gpus=0 --train-steps=1000
```

```
python Collective_worker1.py --data-dir=./cifar-10-data --job-dir=./model_dir/ --num-gpus=0 --train-steps=1000
```

**Other info / logs**

Error message:RecvBufResponse returned 128 bytes where to_tensor expected 64
[[node resnet/BatchNorm/beta/Momentum/Initializer/CollectiveBcastRecv (defined at /Library/Python/2.7/site-packages/tensorflow/contrib/distribute/python/collective_all_reduce_strategy.py:186) ]]

**Comment by byronyi** [comment]
Thanks! It would be better if you could give a minimal reproduce and more verbose logging.

**Comment by poxvoculi** [comment]
An error in CollectiveBcastRecv indicates the problem is probably during model initialization.  The size mismatch suggests the model has been set up differently at each worker.   @yuefengz may be able to pinpoint the problem more exactly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yuefengz: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yuefengz: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yuefengz** [comment]
@CheukNgai Seems your code has logic that only can apply for parameter servers such as `tf.contrib.training.GreedyLoadBalancingStrategy`. Could you remove these logic and try again? It is not recommended to control device placement when you are using distribution strategies.

**Comment by CheukNgai** [comment]
@clustarycx Could you pls help following this issue? If not, i will close this issue because i have left my intern job.

**Comment by clustarycx** [comment]
@CheukNgai I cannot find the recreating procedure. I think you can just closet this, and we will open a new issue or reopen this one if similar problem is hit again. thanks.
